### DISCUSSION TITLE
Add bucket summary in tooltip

### DISCUSSION THREAD

**Discussion by avalara-justin-closs** [marcusolsson/grafana-hourly-heatmap-panel#55] [Ideas]
Hey there,

Feature Request/Implementation question. I am adding an hourly heatmap to a grafana dashboard I am working on, but I have 4 status' of an hourly task that I want to report on a heatmap eg ('done', 'running', 'error', 'pending').

 I was able to implement this by changing those status' to a number, but now I would like to override that number back to the status name proper. Is this something that is currently accessible within the heatmap?

Thanks!

**Comment by marcusolsson** [comment]
Since each bucket is an aggregation of potentially multiple data point, how would this work when a bucket contains tasks with different statuses?

**Comment by avalara-justin-closs** [comment]
That's a good point. In my use case it is not something to be considered as it is a task run on a chron job.

**Comment by marcusolsson** [comment]
Gotcha. I'd consider adding something like this if we can figure out a good user experience. 

My first thought would be to select a field to use as tags to display in the tooltip. As mentioned, this means that we need to figure which tags to display.

- The first/last tag in the bucket
- Tag with the most occurrences
- All unique tags

If we display all unique tags, I see a UX problem if there are, say, 20-30 unique tags in a bucket. We obviously wouldn't want to display all of them. If we limit to display say 5 tags, followed by "... and 56 more tags", I'm not sure whether it would be useful.

I've been thinking about how we'd display annotations in the heatmap, which is a similar problem. What if a bucket contains multiple annotations? If we find a good solution one of the problems, we might be able to use it for both.
